Git-commit: 41a06d113a1b0a25f3964acde5ac68661b4b3056
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 16 Dec 2020 09:28:17 +0000
Subject: [PATCH] overlays: Add missing addresses to ads1015/ads1115
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.5.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

raspberrypi inclusion
category: feature
bugzilla: 50432

--------------------------------

The overlays for the ads1015 and ads1115 I2C ADCs omitted the addresses
in the main device node names. As well as breaking the conventions for
I2C devices, this prevents the firmware from renaming them when the
"reg" property is modified, which in turn stops the overlays from being
instantiated multiple times.

See: https://www.raspberrypi.org/forums/viewtopic.php?f=107&t=294465

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
Signed-off-by: Fang Yafen <yafen@iscas.ac.cn>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 arch/arm/boot/dts/overlays/ads1015-overlay.dts | 2 +-
 arch/arm/boot/dts/overlays/ads1115-overlay.dts | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/ads1015-overlay.dts b/arch/arm/boot/dts/overlays/ads1015-overlay.dts
index deeee1228395..dc1764613a8b 100644
--- a/arch/arm/boot/dts/overlays/ads1015-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ads1015-overlay.dts
@@ -13,7 +13,7 @@ __overlay__ {
             #address-cells = <1>;
             #size-cells = <0>;
             status = "okay";
-            ads1015: ads1015 {
+            ads1015: ads1015@48 {
                 compatible = "ti,ads1015";
                 status = "okay";
                 #address-cells = <1>;
diff --git a/arch/arm/boot/dts/overlays/ads1115-overlay.dts b/arch/arm/boot/dts/overlays/ads1115-overlay.dts
index 4fc571c2db33..e44ced704ee2 100644
--- a/arch/arm/boot/dts/overlays/ads1115-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ads1115-overlay.dts
@@ -15,7 +15,7 @@ __overlay__ {
 			#size-cells = <0>;
 			status = "okay";
 
-			ads1115: ads1115 {
+			ads1115: ads1115@48 {
 				compatible = "ti,ads1115";
 				status = "okay";
 				#address-cells = <1>;
-- 
2.31.1

